### ISSUE TITLE
Inconsistent surfacing of Save and Add Another buttons

### ISSUE THREAD

**Issue by sanjuroj** [mercycorps/TolaActivity#1399]
- In the PA admin Users and Organizations have "Save and add another" buttons but Programs and Countries don't.  
-The Users Save and add another button doesn't result in a blank creation template the way the Organization Save and add another button does
- Save and add another button disappears on Organization form after the first time you've clicked it.

**Comment by jennyhavah** [comment]
> In the PA admin Users and Organizations have "Save and add another" buttons but Programs and Countries don't.

It does seem odd that two have them and two don't. The one justification I can think of is that adding countries and programs is pretty rare, not something you tend to do in rapid succession. Then again, the same could be said of Organizations... How often does that happen. I'd like to ask @hannalcamp to validate whether or not this discrepancy was intentional.

**Comment by jennyhavah** [comment]
> -The  Users Save and add another button doesn't result in a blank creation template the way the Organization Save and add another button does

This definitely looks to be a bug, as the whole point of that button is to open a new form. I was able to replicate it.

**Comment by jennyhavah** [comment]
> Save and add another button disappears on Organization form after the first time you've clicked it.

@sanjuroj Do you mean that when you open up a saved Org profile, that you don't see a "Save and add another Org" button? I believe this is intentional, and it appears to be consistent across sections. I.e. if I add a user and then open the saved user, there's not "Save and add another" button. @hannalcamp can you verify that this is intentional?

**Comment by hannalcamp** [comment]
@jennyhavah Referring to your second comment - when opening a previously-saved profile, it is intentional that the "Save and add another" button is not there. If Sanjuro is seeing that happen even in the blank Add User/Add Organization form, that would be a bug.

In reference to your comment on discrepancy between Add User/Add Organization having the button and Add Program/Add Country not, the discrepancy was not intentional as far as I recall but I see a logic in keeping it, though perhaps putting "Add Program" in the first bucket which actually has a "Save and add another" button. Adding users, organizations, and programs is something I can see Admins doing in batches - Adding countries is something only Super Admins will do and I can't imagine a scenario where batches are needed.

**Comment by RickieT** [comment]
> * In the PA admin Users and Organizations have "Save and add another" buttons but Programs and Countries don't.
>   -The Users Save and add another button doesn't result in a blank creation template the way the Organization Save and add another button does
> * Save and add another button disappears on Organization form after the first time you've clicked it.

The only issue I was able to reproduce here was  "Users add another button does not result in a blank creation template"

Programs and Orgs both have "Save and Add Another" buttons, and they give you a blank form and lets you add another after creating each time.

**Comment by jennyhavah** [comment]
This is what I see when I click "Add program" -- don't see a "Save and add another button.

![image](https://user-images.githubusercontent.com/33670923/56149288-0a40dc00-5f61-11e9-9c23-13e5a80ce906.png)

 I'm stumped as to why Sanjuro and I would not see it and @RickieT and @hannalcamp can.

**Comment by jennyhavah** [comment]
> @sanjuroj @jennyhavah The only issue I was able to reproduce here was "Users add another button does not result in a blank creation template"
Programs and Orgs both have "Save and Add Another" buttons, and they give you a blank form and lets you add another after creating each time. Also the one in the org form never disappeared for me.

I created a new ticket -- #1400 -- for the bug that @RickieT was able to reproduce and moved it into This Release, because it is broken functionality that seems like a totally different issue than whether or not the button appears.

We can keep everything else here.

**Comment by sanjuroj** [comment]
The third potential issue is a little bit different that what you have said you tried so far.  I added reproduction instructions, hopefully that helps.

**Comment by RickieT** [comment]
> The third potential issue is a little bit different that what you have said you tried so far. I added reproduction instructions, hopefully that helps.

@sanjuroj Yeah, I understand this issue. If you click outside the form on a previously saved program, the "save and add another" option is gone, which I don't know that I'd consider a bug rather than a usability problem. 

If you create a new program and then click save and add another, you can do this multiple times, as you are in a creation form. Once you click outside the form I would assume you're now viewing/editing existing orgs.

**Comment by sanjuroj** [comment]
Weird, I can't reproduce that third one now either. So I guess it's fine.

**Comment by jennyhavah** [comment]
I agree that we want a "Save and add another button" if you're in the process of adding new Orgs, Users, whatever.

I do NOT think it makes sense to include the "Save and add another" button if you are opening up an already-saved element, to edit it.  "Add another" implies that you just added something. It seems out of context if you're just editing something.

**Comment by jennyhavah** [comment]
Looks like the only thing remaining in this ticket is to add a "Save and add another" button to the "Add a program" form. I've updated the original ticket and am moving to Deferred. @hannalcamp if I misunderstood and you actually consider it a release blocker, then of course go ahead and update the ticket accordingly.

**Comment by hannalcamp** [comment]
Nope, I'm good.
